doc/index.md

```diff
@@ -0,0 +1,3 @@
+# Horizen Sidechain SDK Release Notes
+
+## [EON 1.0.0](/doc/release/eon_1.0.0.md) (2023-09-06)
```

Comment:
let's remove any reference to EON, here we are in the sdk
this release note is for 0.8.0